src/object.c

```diff
@@ -208,10 +209,56 @@ int write_object_file(SDL_RWops *rw, rct_object_entry* entry)
 	chunkHeader.encoding = object_entry_group_encoding[type];
 	chunkHeader.length = installed_entry->chunk_size;
 
+
+	//Check if content of object file matches the stored checksum. If it does not, then fix it.
+	int calculated_checksum=object_calculate_checksum(entry,chunk,installed_entry->chunk_size);
+		if(entry->checksum!=calculated_checksum)
+		{
+		printf("Chunk size %d\n",chunkHeader.length);
+
+		printf("Fixing object with invalid checksum\n");
+		//Store the current length of the header - it's the offset at which we will write the extra bytes
+		int salt_offset=chunkHeader.length;
+		/*Allocate a new chunk 11 bytes longer.
+		I would just realloc the old one, but realloc can move the data, leaving dangling pointers 
+		into the old buffer. If the chunk is only referenced in one place it would be safe to realloc 
+		it and update that reference, but I don't know the codebase well enough to know if that's the
+		case, so to be on the safe side I copy it*/
+		uint8* new_chunk=malloc(chunkHeader.length+11);
```

Comment:
Oops... because I initially used realloc(), I forgot to add it. Should I submit a different pull request or is there a way to change this one?